### PR TITLE
[MOO-897] React 18 and RN 0.70

### PR DESCRIPTION
**Pull request by diego-antonelli**
## Checklist

-   Contains unit tests ✅
-   Contains breaking changes ✅
-   Compatible with: MX 9️⃣/🔟
-   Did you update version and changelog? ✅
-   PR title properly formatted (`[XX-000]: description`)? ✅

## This PR contains

-   [ ] Bug fix
-   [x] Feature
-   [ ] Refactor
-   [ ] Documentation
-   [ ] Other (describe)

## What is the purpose of this PR?
Introduce updates for React 18 and React Native 0.71, this also introduces Jest 29 and new test configurations.

## Relevant changes

Libraries and unit tests.

## What should be covered while testing?

Backwards compatibility is not something applicable here.

### CODE REVIEW

File: packages/pluggable-widgets-tools/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mendix/pluggable-widgets-tools",
-  "version": "9.20.0",
+  "version": "9.23.0",
```
Comment (tamelbrun) [inline packages/pluggable-widgets-tools/package.json:3]: Note to self: this will become 9.24 instead.

Comment (tamelbrun) [reply packages/pluggable-widgets-tools/package.json]: The version is now set at `9.24.0-beta.1` for this branch, it should/will be changed to 9.24.0 when we are close to merging/releasing.

Comment (diego-antonelli) [reply packages/pluggable-widgets-tools/package.json]: Done

File: packages/pluggable-widgets-tools/test-config/jest.config.js [hunk 2]
```diff
@@ -4,27 +4,25 @@ const projectDir = process.cwd();
...
+        "^.+\\.jsx?$": join(__dirname, "transform.js"),
+        "^.+\\.svg$": join(__dirname, "jest-svg-transformer")
     },
     moduleNameMapper: {
         "\\.(css|less|scss|sass)$": "identity-obj-proxy",
         "mendix/components/web/Icon": join(__dirname, "__mocks__/WebIcon"),
         "mendix/filters/builders": join(__dirname, "__mocks__/FilterBuilders"),
         "\\.png$": join(__dirname, "assetsTransformer.js")
     },
+    moduleDirectories: ["node_modules", join(projectDir, "node_modules"), join(__dirname, "../../../")],
```
Comment (tamelbrun) [inline packages/pluggable-widgets-tools/test-config/jest.config.js:24]: This triple `..` looks a bit fishy ;).

**Pull request closed without merging**